pylint/checkers/format.py

```diff
@@ -1231,41 +1233,64 @@ def _check_multi_statement_line(self, node, line):
         self.add_message("multiple-statements", node=node)
         self._visited_lines[line] = 2
 
-    def check_lines(self, lines, i):
-        """check lines have less than a maximum number of characters
+    def check_line_ending(self, line: str, i: int) -> None:
+        """
+        Check that the final newline is not missing and that there is no trailing whitespace.
+        """
+        if not line.endswith("\n"):
+            self.add_message("missing-final-newline", line=i)
+        else:
+            # exclude \f (formfeed) from the rstrip
+            stripped_line = line.rstrip("\t\n\r\v ")
+            if not stripped_line and _EMPTY_LINE in self.config.no_space_check:
+                # allow empty lines
+                pass
+            elif line[len(stripped_line) :] not in ("\n", "\r\n"):
+                self.add_message(
+                    "trailing-whitespace", line=i, col_offset=len(stripped_line)
+                )
+            # Don't count excess whitespace in the line length.
+            line = stripped_line
```

Comment:
Trivial: Looks like this bit isn't needed anymore?

Reply:
Thanks @AWhetter !
It was indeed dead code that i forgot to delete.